src/submatrix_methods.F

```diff
@@ -0,0 +1,112 @@
+!--------------------------------------------------------------------------------------------------!
+!   CP2K: A general program to perform molecular dynamics simulations                              !
+!   Copyright (C) 2000 - 2020  CP2K developers group                                               !
+!--------------------------------------------------------------------------------------------------!
+
+MODULE submatrix_methods
+
+   IMPLICIT NONE
+
+CONTAINS
+
+! **************************************************************************************************
+!> \brief sort integer array using quicksort
+!> \param arr_a - input array
+!> \param left - left boundary of region to be sorted
+!> \param right - right boundary of region to be sorted
+! **************************************************************************************************
+   RECURSIVE PURE SUBROUTINE qsort(arr_a, left, right)
```

Comment:
I think we have already some sorting routines (module util, routine sort)

Reply:
Thanks for the hint! I now replaced qsort by sort from util. However, qsort_two behaves differently as it sorts two lists simultaneously and uses the content of both to determine the order. Therefore I kept qsort_two for now.